### ISSUE TITLE
Uninitialized read in FileDialog.cpp

### ISSUE THREAD

**Issue by ariccio** [notepad-plus-plus/notepad-plus-plus#514]
[As I alluded to in my rejected PR](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/180/files#diff-52fceea5a45c6c87de189239d3874211R225), #180 , there's an uninitialized read [in `FileDialog::doOpenMultiFilesDlg`, variable `fn`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/WinControls/OpenSaveFileDialog/FileDialog.cpp#L225):

```
		TCHAR fn[MAX_PATH];
		TCHAR *pFn = _fileName + lstrlen(_fileName) + 1;
		if (!(*pFn))
			_fileNames.push_back(generic_string(_fileName));
		else
		{
			lstrcpy(fn, _fileName);
			if (fn[lstrlen(fn)-1] != '\\')
				lstrcat(fn, TEXT("\\"));
		}
		int term = int(lstrlen(fn));
```
...aside from the egregious use of very dangerous string handling functions, in the `if (!(*pFn))` code path, `int term = int(lstrlen(fn))` reads garbage memory.

Also: `_fileName` is declared:

```
TCHAR _fileName[MAX_PATH*8];
```

....which means that in the *NON*-`if (!(*pFn))` (the `else`) code path, long file names corrupt the stack! Using `_tcscpy_s` and friends would avoid stack corruption, with no code changes necessary.

**Comment by pnedev** [comment]
Good catch!

**Comment by ariccio** [comment]
Fixed in fe2defb53664849c394a76ea2bfd2dae36c7d00d
